src/main/java/picard/sam/SplitSamBySize.java

```diff
@@ -0,0 +1,152 @@
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package picard.sam;
+
+import htsjdk.samtools.*;
+import htsjdk.samtools.util.IOUtil;
+import htsjdk.samtools.util.Log;
+import htsjdk.samtools.util.ProgressLogger;
+import org.broadinstitute.barclay.argparser.Argument;
+import org.broadinstitute.barclay.argparser.CommandLineProgramProperties;
+import org.broadinstitute.barclay.help.DocumentedFeature;
+import picard.PicardException;
+import picard.cmdline.CommandLineProgram;
+import picard.cmdline.StandardOptionDefinitions;
+import picard.cmdline.programgroups.SamOrBam;
+
+import java.io.File;
+import java.text.DecimalFormat;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * <p/>
+ * Splits the input queryname sorted or query-grouped SAM/BAM file and writes it into
+ * multiple BAM files, each with an approximately equal number of reads. This will retain the sort order
+ * within each output BAM and if the BAMs are concatenated in order (output files are named
+ * numerically) the order of the reads will match the original BAM.
+ */
+@CommandLineProgramProperties(
+        summary = SplitSamBySize.USAGE_SUMMARY + SplitSamBySize.USAGE_DETAILS,
+        oneLineSummary = SplitSamBySize.USAGE_SUMMARY,
+        programGroup = SamOrBam.class)
+@DocumentedFeature
+public class SplitSamBySize extends CommandLineProgram {
+    static final String USAGE_SUMMARY = "Splits a SAM or BAM file to multiple BAMs.";
+    static final String USAGE_DETAILS = "This tool splits the input query-grouped SAM/BAM file into multiple BAM files " +
+            "while maintaining the sort order. This can be used to split a large unmapped BAM in order to parallelize alignment."+
+            "<br />" +
+            "<h4>Usage example:</h4>" +
+            "<pre>" +
+            "java -jar picard.jar SplitSamBySize \\<br />" +
+            "     I=paired_unmapped_input.bam \\<br />" +
+            "     OUTPUT_DIR=out_dir \\ <br />" +
+            "     TOTAL_READS_IN_INPUT=800000000 \\ <br />" +
+            "     SPLIT_TO_N_READS=48000000" +
+            "</pre>" +
+            "<hr />";
+    @Argument(doc = "Input SAM/BAM file to split", shortName = StandardOptionDefinitions.INPUT_SHORT_NAME)
+    public File INPUT;
+
+    @Argument(shortName = "N_READS", doc = "Split to have approximately N reads per output file.", optional = true,
+            mutex = {"SPLIT_TO_N_FILES"})
+    public int SPLIT_TO_N_READS;
+
+    @Argument(shortName = "N_FILES", doc = "Split to N files.", optional = true, mutex = {"SPLIT_TO_N_READS"})
+    public int SPLIT_TO_N_FILES;
+
+    @Argument(shortName = "TOTAL_READS", doc = "Total number of reads in the input file.")
+    public int TOTAL_READS_IN_INPUT;
+
+    @Argument(shortName = "O", doc = "Directory in which to output the split BAM files.")
+    public File OUTPUT_DIR;
+
+    @Argument(shortName = "OUT_PREFIX", doc = "Prefix to name output files.")
```

Comment:
make the doc more explicit..."Output files will be named <OUT_PREFIX>_N.bam, where N enumerates the output file."